### PR TITLE
Add an unit test in JMSSenderTestCase

### PR DESCRIPTION
**Pull request by tharinduwijewardane**
Test case testGettingHyphenModeFromJMSConnectionFactory added to cover [this](https://github.com/wso2/wso2-axis2-transports/pull/194) bugfix

### CODE REVIEW

File: modules/jms/src/test/java/org/apache/axis2/transport/jms/JMSSenderTestCase.java [hunk 1]
```diff
@@ -133,6 +133,38 @@ public void testGetContentTypePropertyNameFromFactory() {
         Assert.assertEquals(contentTypeProperty, contentTypePropertyResult);
     }
 
+    public void testGettingHyphenModeFromJMSConnectionFactory() throws Exception {
```
Comment (ThishaniLucas) [inline modules/jms/src/test/java/org/apache/axis2/transport/jms/JMSSenderTestCase.java]: Shall we add a test method description?

Comment (tharinduwijewardane) [reply modules/jms/src/test/java/org/apache/axis2/transport/jms/JMSSenderTestCase.java]: I think the method name is quite descriptive

File: modules/jms/src/test/java/org/apache/axis2/transport/jms/JMSSenderTestCase.java [hunk 2]
```diff
@@ -133,6 +133,38 @@ public void testGetContentTypePropertyNameFromFactory() {
...
+        PowerMockito.doReturn(jmsConnectionFactory).when(jmsSender, "getJMSConnectionFactory", ArgumentMatchers.any());
+        PowerMockito.doReturn(destination)
+                    .when(jmsSender, "sendOverJMS", ArgumentMatchers.any(), ArgumentMatchers.any(),
+                          ArgumentMatchers.any(), ArgumentMatchers.any(), ArgumentMatchers.any());
+
+        jmsSender.init(new ConfigurationContext(new AxisConfiguration()), new TransportOutDescription("jms"));
+        MessageContext messageContext = new MessageContext();
+        jmsSender.sendMessage(messageContext, "jms:/SimpleStockQuoteService", null);
+
+        Assert.assertEquals("Hyphen mode not set", hyphenMode,
```
Comment (malakaganga) [inline modules/jms/src/test/java/org/apache/axis2/transport/jms/JMSSenderTestCase.java]: Shall we add a failure message?

File: modules/jms/src/test/java/org/apache/axis2/transport/jms/JMSSenderTestCase.java (hunk 1 above)
Comment (tharinduwijewardane) [reply modules/jms/src/test/java/org/apache/axis2/transport/jms/JMSSenderTestCase.java]: added a javadoc comment

File: modules/jms/src/test/java/org/apache/axis2/transport/jms/JMSSenderTestCase.java (hunk 2 above)
Comment (tharinduwijewardane) [reply modules/jms/src/test/java/org/apache/axis2/transport/jms/JMSSenderTestCase.java]: Improved the failure message.

**Pull request merged**